### PR TITLE
fix(formal-core): formal.change on multiple fields fails

### LINKED ISSUE #33: Using `formal.change` on multiple fields in succession fails to update
# Bug

I have a geosearch component I use outside of formal to 'inject' address information into a form. When the user completes typing in an address, I call `formal.change` several times in one go in order to change these fields individually.

Unfortunately, there is a bug in the `change` handler implementation in `formal-core`, which causes none of the fields to update at all. This is related to the way hooks work and how state in `useState` is updated. 

## Reproduction steps

N/A, see fork for updated test which fails on non-bugfixed code

## Suggested solution

I have an updated test + a fix ready in a PR, which uses the `setState((oldState) => {...oldState, [key]: value}` notation instead of `setState({...values, [key]: value)` which doesn't work in this case. Added benefit: the callback dependency on `[values]` is no longer needed. 

See PR (will be added in edit shortly)

### PR DESCRIPTION
**Pull request by hilkeheremans**
# Whats new?

A fix for a bug where multiple `formal.change` calls in succession would fail to update all fields.

## Issue link

Closes #33 

## Notes

N/A

### CODE REVIEW

**Comment by hilkeheremans** [comment]
Just as an aside, I have had lots of issues running `yarn validate`, even with a clean `git clone` -- I'm guessing some of the linting dependencies have been updated/improved and are screaming about linting issues that were previous not there...

**Comment by hilkeheremans** [comment]
FYI: I have covered the bug in an existing testcase, so this does not affect coverage, nor does it add an extra test. If you prefer splitting off this exact case in a separate test, let me know and I'll do what's needed.

File: packages/formal/src/use-formal.ts [hunk 1]
```diff
@@ -33,9 +33,9 @@ export default function useFormal<Schema>(
 
   const change = useCallback(
     (field: keyof Schema, value: any): void => {
-      setValues({ ...values, [field]: value })
+      setValues((prevValues: any) => ({ ...prevValues, [field]: value }))
```
Comment (kevinwolfcr) [inline packages/formal/src/use-formal.ts]: Can you just type `prevValues` with `Schema` rather than `any`, please?

**Comment by kevinwolfcr** [comment]
@hilkeheremans regarding eslint issues, it is because of a bad version of `eslint-plugin-import`. I will update dependencies tomorrow according to https://github.com/kevinwolfcr/formal/issues/32#issuecomment-482835733

**Comment by hilkeheremans** [comment]
There you go!

File: packages/formal/src/use-formal.ts (hunk 1 above)
Comment (hilkeheremans) [reply packages/formal/src/use-formal.ts]: See below

**Comment by kevinwolfcr** [comment]
Merged!

**Comment by kevinwolfcr** [comment]
@allcontributors please add @hilkeheremans as a contributor for bugfix

**Comment by kevinwolfcr** [comment]
@allcontributors please add @hilkeheremans for bug and code

**Pull request merged**